### PR TITLE
[Hotfix][e2e] fix JobRestoreWhenMasterNodeSwitch method NPE

### PR DESCRIPTION
**Pull request by TaoZex**
## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

File: seatunnel-engine/seatunnel-engine-server/src/test/java/org/apache/seatunnel/engine/server/CoordinatorServiceTest.java [hunk 1]
```diff
@@ -154,7 +154,7 @@ public void testJobRestoreWhenMasterNodeSwitch() throws InterruptedException {
             });
 
         // wait job restore
-        Thread.sleep(5000);
+        Thread.sleep(30000);
```
Comment (CalvinKirs) [inline seatunnel-engine/seatunnel-engine-server/src/test/java/org/apache/seatunnel/engine/server/CoordinatorServiceTest.java]: This is not a good practice. @Hisoka-X do you have any good ideas?

Comment (Hisoka-X) [reply seatunnel-engine/seatunnel-engine-server/src/test/java/org/apache/seatunnel/engine/server/CoordinatorServiceTest.java]: If we can catch log to make sure job restore success, maybe we can remove this code. But now, we don't have good way.

File: seatunnel-engine/seatunnel-engine-server/src/test/java/org/apache/seatunnel/engine/server/CoordinatorServiceTest.java [hunk 2]
```diff
@@ -154,7 +154,7 @@ public void testJobRestoreWhenMasterNodeSwitch() throws InterruptedException {
...
 
         // pipeline will recovery running state
         await().atMost(200000, TimeUnit.MILLISECONDS)
```
Comment (FWLamb) [inline seatunnel-engine/seatunnel-engine-server/src/test/java/org/apache/seatunnel/engine/server/CoordinatorServiceTest.java:164]: Increasing the waiting time here may succeed, For example, 10 minutes.

File: seatunnel-engine/seatunnel-engine-server/src/test/java/org/apache/seatunnel/engine/server/CoordinatorServiceTest.java (hunk 1 above)
Comment (TaoZex) [reply seatunnel-engine/seatunnel-engine-server/src/test/java/org/apache/seatunnel/engine/server/CoordinatorServiceTest.java]: Agree with you, I think the problem is the code getJobMaster(jobId), which returns null when the master has not successfully switched.
If we can't make sure job restore success now, we can either temporarily disable it or set the code Thread.sleep to a large value.

Comment (EricJoy2048) [reply seatunnel-engine/seatunnel-engine-server/src/test/java/org/apache/seatunnel/engine/server/CoordinatorServiceTest.java]: `Thread.sleep(5000) ` is used to wait the job begin restore and job status turn to a non `RUNNING` status. And then I use 

```
// pipeline will recovery running state
        await().atMost(200000, TimeUnit.MILLISECONDS)
            .untilAsserted(
                () -> Assertions.assertEquals(PipelineStatus.RUNNING,
                    server2.getCoordinatorService().getJobMaster(jobId).getPhysicalPlan().getPipelineList().get(0)
                        .getPipelineState()));

```

to ensure the job restore complete and job status turn to running again.
The problem is we can not ensure the job is begin restore and the status turn to a non `RUNNING` status. I suggest use `await().atMost()` to check the job status leave `RUNNING`.

**Review by Hisoka-X — APPROVED** [review]
LGTM

**Review by EricJoy2048 — APPROVED** [review]
LGTM

**Comment by hailin0** [comment]
LGTM

**Pull request merged**